compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -97,12 +100,21 @@ pub fn link_binary<'a>(
                 .tempdir()
                 .unwrap_or_else(|error| sess.emit_fatal(errors::CreateTempDir { error }));
             let path = MaybeTempDir::new(tmpdir, sess.opts.cg.save_temps);
-            let out_filename = out_filename(
+            let output = out_filename(
                 sess,
                 crate_type,
                 outputs,
                 codegen_results.crate_info.local_crate_name,
             );
+            let out_filename = match output {
```

Comment:
This could be included in a `OutputFileName::New` function.

Reply:
Do you mean defining something like this?

```rust
impl OutFileName {
    pub fn temp_path(&self, flavor: OutputType, codegen_unit_name: Option<&str>) -> PathBuf {
        OutFileName::Real(ref path) => path.clone(),
        OutFileName::Stdout => outputs.temp_path(flavor, codegen_unit_name),
    }
}
```

Then use it here:
```rust
let out_filename = output.temp_path();
```

Reply:
Seems good. For the part before linking you could also have a method like `get_filename` on `OutFileName`, can't you?

Reply:
Yes I can. I added two helpers to `OutFileName` and simplified the part before linking. IMHO, I'm leaning towards keeping the after-linking part as is, because - in addition to my point above - the logic is one step of `link_binary`.